File: codegen/results/v002/qwen3.5_27b_nothink/20260411_193102/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
	"strconv"
)

type Person struct {
	Name  string  `json:"Name"`
	Age   int     `json:"Age"`
	Email string  `json:"Email"`
	Score float64 `json:"Score"`
}

func main() {
	// Open the file
	file, err := os.Open("input/data.csv")
	if err != nil {
		fmt.Println("Error opening file:", err)
		os.Exit(1)
	}
	defer file.Close()

	// Create a CSV reader
	reader := csv.NewReader(file)

	// Read all records
	records, err := reader.ReadAll()
	if err != nil {
		fmt.Println("Error reading CSV:", err)
		os.Exit(1)
	}

	// Skip header (first row)
	if len(records) < 2 {
		// Output empty JSON array if no data rows
		fmt.Println("[]")
		return
	}

	var people []Person

	// Iterate over data rows (starting from index 1)
	for _, row := range records[1:] {
		if len(row) < 4 {
			continue // Skip malformed rows
		}

		age, err := strconv.Atoi(row[1])
		if err != nil {
			fmt.Println("Error parsing Age:", err)
			continue
		}

		score, err := strconv.ParseFloat(row[3], 64)
		if err != nil {
			fmt.Println("Error parsing Score:", err)
			continue
		}

		person := Person{
			Name:  row[0],
			Age:   age,
			Email: row[2],
			Score: score,
		}

		people = append(people, person)
	}

	// Marshal to JSON with indentation
	jsonOutput, err := json.MarshalIndent(people, "", "  ")
	if err != nil {
		fmt.Println("Error marshaling JSON:", err)
		os.Exit(1)
	}

	// Write to stdout
	fmt.Println(string(jsonOutput))
}

File: codegen/results/v002/qwen3.5_27b_nothink/20260411_193102/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

interface DataEntry {
  Name: string;
  Age: number;
  Email: string;
  Score: number;
}

function main() {
  const filePath = path.join(process.cwd(), 'input', 'data.csv');
  
  try {
    const content = fs.readFileSync(filePath, 'utf-8');
    const lines = content.trim().split('\n');

    if (lines.length < 2) {
      console.log('[]');
      return;
    }

    // Skip header (first line)
    const dataLines = lines.slice(1);
    
    const result: DataEntry[] = dataLines.map(line => {
      // Handle CSV parsing that might have commas within quoted fields
      // For this specific input format, simple split is sufficient as there are no commas in fields
      const parts = line.split(',');
      
      if (parts.length !== 4) {
        throw new Error(`Invalid CSV line format: ${line}`);
      }

      return {
        Name: parts[0],
        Age: parseInt(parts[1], 10),
        Email: parts[2],
        Score: parseFloat(parts[3])
      };
    });

    // Output as valid JSON to stdout
    console.log(JSON.stringify(result, null, 2));
  } catch (error) {
    console.error('Error:', (error as Error).message);
    process.exit(1);
  }
}

main();